_posts/2021-01-13-hapi-fhir.md

```diff
@@ -0,0 +1,454 @@
+
+- Hapi-fhir-server
+
+  
+
+pom.xml
+
+```xml
+<dependencies>
+    <dependency>
+        <groupId>ca.uhn.hapi.fhir</groupId>
+        <artifactId>hapi-fhir-structures-r4</artifactId>
+        <version>5.2.0</version>
+    </dependency>
+    <dependency>
+        <groupId>ca.uhn.hapi.fhir</groupId>
+        <artifactId>hapi-fhir-base</artifactId>
+        <version>5.2.0</version>
+    </dependency>
+    <dependency>
+        <groupId>ca.uhn.hapi.fhir</groupId>
+        <artifactId>hapi-fhir-server</artifactId>
+        <version>5.2.0</version>
+    </dependency>
+</dependencies>
+```
+
+### Server
+
+This class is going to be the heart of your FHIR code. 
+Here you will configure most of your HAPI server settings. 
+You can add security, select which resources your server will support, 
+set defaults on the server to use XML or JSON and many more options. We will implement some more later in this post.
+
+FhirRestfullServer.java
+
+```Java
+import ca.uhn.fhir.context.FhirContext;
+import ca.uhn.fhir.rest.server.RestfulServer;
+
+import com.example.fhirexample.providors.ObservationProvidor;
+import com.example.fhirexample.providors.PatientProvidor;
+
+import org.springframework.context.ApplicationContext;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import java.util.Arrays;
+
+@WebServlet("/*")
+public class FhirRestfullServer extends RestfulServer {
+
+    private ApplicationContext applicationContext;
+
+    FhirRestfullServer(ApplicationContext context) {
+
+        this.applicationContext = context;
+    }
+
+    @Override
+    protected void initialize() throws ServletException{
+        super.initialize();
+        setFhirContext(FhirContext.forR4());
+    }
+}
+```
+
+Now lets setup the Servlet Context. 
+This is Spring Boot ABC. Pay attention to the Bean Annotation. 
+This is where we are bringing FHIR into play.
+
+```Java
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.web.servlet.ServletRegistrationBean;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.Bean;
+
+@SpringBootApplication
+public class FhirexampleApplication {
+
+    @Autowired
+    private ApplicationContext context;
+
+
+    public static void main(String[] args) {
+
+        SpringApplication.run(FhirexampleApplication.class, args);
+    }
+
+    @Bean
+    public ServletRegistrationBean ServletRegistrationBean() {
+        ServletRegistrationBean registration= new ServletRegistrationBean(new FhirRestfullServer(context),"/*");
+        registration.setName("FhirServlet");
+        return registration;
+    }
+}
+```
+
+Once your server has started, 
+Open up your Postman and enter this URL http://localhost:8080/metadata. 
+A capability statement will appear just like the screenshot below. 
```

Comment:
Maybe you can elaborate on what a "capability statement" is: A Capability Statement documents a set of capabilities (behaviors) of a FHIR Server for a particular version of FHIR that may be used as a statement of actual server functionality or a statement of required or desired server implementation.